src/mongo/Config.class.php

```diff
@@ -1759,7 +1760,23 @@ protected function getConnectionForDataSource($dataSource)
         }
         if(!isset($this->connections[$dataSource]))
         {
-            $this->connections[$dataSource] = new \MongoClient($ds['connection'], $connectionOptions);
+            $retries = 1;
+            $exception = null;
+
+            do {
+                try {
+                    $this->connections[$dataSource] = new \MongoClient($ds['connection'], $connectionOptions);
+                } catch (\MongoConnectionException $e) {
```

Comment:
There is one error code here - 71 - but you get that for every mongo connection exception so i didn't track for one specifically.